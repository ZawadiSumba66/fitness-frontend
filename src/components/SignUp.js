import { connect } from 'react-redux';
import { useState } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';
import createUser from '../actions/user_action';

const SignUp = ({ newUser, isLogin, history }) => {
  const [userName, setUserName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordConfirmation, setPasswordConfirmation] = useState('');
  const [avatar, setAvatar] = useState('');
  const [message, setMessage] = useState('');

  const uploadFile = async (imageFile) => {
    const formData = new FormData();
    formData.append('file', imageFile);
    formData.append('upload_preset', 'hn9aamrp');
    const response = await axios({
      url: 'https://api.cloudinary.com/v1_1/db69dlggn/image/upload',
      method: 'POST',
      data: formData,
    });
    return response.data.secure_url;
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const imgPath = await uploadFile(avatar);
    await newUser({
      userName, email, password, passwordConfirmation, imgPath,
    });
    if (isLogin === true) {
      history.push('/');
    } else {
      setMessage('welcome');
    }
    return message;
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={userName}
        onChange={(e) => setUserName(e.target.value)}
        type="text"
        name="username"
        placeholder="Enter your username"
      />
      <input
        value={email}
        onChange={(e) => setEmail(e.target.value)}
        type="email"
        name="email"
        placeholder="Enter your email"
      />
      <input
        value={password}
        onChange={(e) => setPassword(e.target.value)}
        type="password"
        name="password"
        placeholder="Password"
      />
      <input
        value={passwordConfirmation}
        onChange={(e) => setPasswordConfirmation(e.target.value)}
        type="password"
        name="passwordconfirmation"
        placeholder="Confirm your password"
      />
      <input
        onChange={(e) => setAvatar(e.target.files[0])}
        type="file"
        name="avatar"
      />
      <button type="submit">
        Create Account
      </button>
    </form>
  );
};

const mapStateToProps = (state) => ({
  user: state.authReducer.user,
  isLogin: state.authReducer.isLogin,
});
const mapDispatchToProps = (dispatch) => ({
  newUser: (estate) => dispatch(createUser(estate)),
});

SignUp.propTypes = {
  newUser: PropTypes.func.isRequired,
  history: PropTypes.instanceOf(Array).isRequired,
  isLogin: PropTypes.bool.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(SignUp);
