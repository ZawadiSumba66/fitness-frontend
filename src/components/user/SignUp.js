import { connect } from 'react-redux';
import { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from '@reach/router';
import { signupUser } from '../../actions/user_action';
import store from '../../store';

const SignUp = ({ error }) => {
  const [userName, setUserName] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordConfirmation, setPasswordConfirmation] = useState('');
  const [avatar, setAvatar] = useState(null);

  const handleFileUpload = (e) => {
    setAvatar(e.target.files[0]);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (avatar !== '') {
      // const formData = new FormData();
      // formData.append('user[username]', userName);
      // formData.append('user[email]', email);
      // formData.append('user[password]', password);
      // formData.append('user[password_confirmation]', passwordConfirmation);
      // formData.append('user[avatar]', avatar);
      const user = {
        username: userName,
        email,
        password,
        password_confirmation: passwordConfirmation,
        avatar,
      };
      console.log(user);
      store.dispatch(signupUser(user));
    } else {
      window.alert(error);
    }
  };

  return (

    <div>
      <form onSubmit={handleSubmit}>
        <input
          value={userName}
          onChange={(e) => setUserName(e.target.value)}
          type="text"
          name="username"
          placeholder="Enter your username"
        />
        <input
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          name="email"
          placeholder="Enter your email"
        />
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          name="password"
          placeholder="Password"
        />
        <input
          value={passwordConfirmation}
          onChange={(e) => setPasswordConfirmation(e.target.value)}
          type="password"
          name="passwordconfirmation"
          placeholder="Confirm your password"
        />
        <input
          onChange={handleFileUpload}
          type="file"
          name="avatar"
        />
        <input
          type="submit"
          className="main-btn primary-shadow"
          value="Create Account"
        />
      </form>
      <p>Have an account?</p>
      <Link
        to="/login"
      >
        Login
      </Link>
    </div>
  );
};

// const mapDispatchToProps = (dispatch) => ({
//   signup: () => dispatch(signupUser),
// });

const mapStateToProps = (state) => ({
  error: state.userReducer.signup_error,
});

SignUp.propTypes = {
  // signup: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired,
};

export default connect(mapStateToProps)(SignUp);
