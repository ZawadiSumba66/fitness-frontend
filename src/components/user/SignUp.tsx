import { connect } from 'react-redux';
import React, { useState } from 'react';
import { Link } from '@reach/router';
import { signupUser } from '../../actions/user_action';
import store from '../../store';
import Flash from './Flash';

type UserProps = {
  backend: string[];
};

const SignUp = ({ backend }: UserProps) => {
  const [username, setUsername] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordConfirmation, setPasswordConfirmation] = useState('');
  const [isLoading, setLoading] = useState(false);
  // const [image, setImage] = useState(null);

  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    // const formData = new FormData();
    // formData.append('username', username);
    // formData.append('email', email);
    // formData.append('password', password);
    // formData.append('password_confirmation', passwordConfirmation);
    // formData.append('image', image);
    const user = {
      username,
      email,
      password,
      password_confirmation: passwordConfirmation,
    } as const;
    store.dispatch(signupUser(user));
    console.log(backend);
    setLoading(false);
  };

  return (

    <div className="flex flex-1 flex-column signup-form px-10 md:px-60 fitness-image">
      <div>
        <i className="fas fa-heartbeat" />
        <span className="font-bold">KEEP IT FIT</span>
      </div>
      <p className="text-white font-bold pb-3"> Hi there! Sign up and start looking for fitness healthy tips that you can practise</p>
      <Flash />
      <div className="errors">
        {backend ? (
          <div>
            {backend.map((item: string) => (
              <li className="text-danger" key={Date.now() * Math.random()}>{item}</li>
            ))}
          </div>
        ) : (
          ''
        )}
      </div>
      <form onSubmit={handleSubmit}>
        <input
          value={username}
          onChange={(e) => setUsername(e.target.value)}
          type="text"
          name="username"
          placeholder="Enter your username"
          className="form-control"
        />
        <input
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          name="email"
          placeholder="Enter your email"
          className="form-control mt-3"
        />
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          name="password"
          placeholder="Password"
          className="form-control mt-3"
        />
        <input
          value={passwordConfirmation}
          onChange={(e) => setPasswordConfirmation(e.target.value)}
          type="password"
          name="password_confirmation"
          placeholder="Confirm your password"
          className="form-control mt-3"
        />
        {/* <input
          type="file"
          name="image"
          onChange={(e) => setImage(e.target.files[0])}
        /> */}
        <br />
        <button
          type="submit"
          className="button-orange btn text-light text-uppercase font-weight-bold w-100 mt-4"
          value="Sign up"
        >
          {isLoading ? 'Loading...' : 'Sign Up'}
        </button>
      </form>
      <p className="pt-5 text-center text-white">Have an account?</p>
      <Link
        to="/login"
        className="text-center text-dark font-bold"
      >
        Login
      </Link>
    </div>
  );
};

type SignupState = {
  userReducer: {
    signup_backend_error: string[]
  }
};

const mapStateToProps = (state: SignupState) => ({
  backend: state.userReducer.signup_backend_error,
});

export default connect(mapStateToProps)(SignUp);
