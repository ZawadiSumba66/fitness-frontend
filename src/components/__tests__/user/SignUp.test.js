import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../../store';
import SignUp from '../../user/SignUp';

it('matches the signup page snapshot', () => {
  const tree = renderer.create(<Provider store={store}><SignUp /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
