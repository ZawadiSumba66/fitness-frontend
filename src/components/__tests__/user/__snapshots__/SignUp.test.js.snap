// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the signup page snapshot 1`] = `
<div
  className="flex flex-1 flex-column signup-form px-10 md:px-60 fitness-image"
>
  <div>
    <i
      className="fas fa-heartbeat"
    />
    <span
      className="font-bold"
    >
      KEEP IT FIT
    </span>
  </div>
  <p
    className="text-white font-bold pb-3"
  >
     Hi there! Sign up and start looking for fitness healthy tips that you can practise
  </p>
  <div />
  <div
    className="errors"
  >
    <div />
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      className="form-control"
      name="username"
      onChange={[Function]}
      placeholder="Enter your username"
      type="text"
      value=""
    />
    <input
      className="form-control mt-3"
      name="email"
      onChange={[Function]}
      placeholder="Enter your email"
      type="email"
      value=""
    />
    <input
      className="form-control mt-3"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="form-control mt-3"
      name="password_confirmation"
      onChange={[Function]}
      placeholder="Confirm your password"
      type="password"
      value=""
    />
    <label
      htmlFor="avatar"
    >
      Upload Avatar
      <input
        className="w-100 my-3"
        id="avatar"
        onChange={[Function]}
        type="file"
      />
    </label>
    <br />
    <button
      className="button-orange btn text-light text-uppercase font-weight-bold w-100 mt-4"
      type="submit"
      value="Sign up"
    >
      SignUp
    </button>
  </form>
  <p
    className="pt-5 text-center text-white"
  >
    Have an account?
  </p>
  <a
    className="text-center text-dark font-bold"
    href="/login"
    onClick={[Function]}
  >
    Login
  </a>
</div>
`;
